scintillometry/io/psrfits/core.py

```diff
@@ -61,7 +63,37 @@ def open(filename, mode='r', **kwargs):
         # TODO: allow support for more than one HDU.
         if len(reader_list) != 1:
             raise RuntimeError("Current reader can only read one SUBINT HDU.")
+        # TODO Returning one HDU for now. But in the future, we may want to
+        # return a group of it.
         return reader_list[0]
+
+    elif mode in ['w', 'rb+']:
+        # Check if file exists
+        is_file = os.path.exists(filename)
+        if is_file and not overwrite:
+            raise RuntimeError("file '{}' exists. Use 'overwrite=Ture' to "
+                               "overwrite it.".format(filename))
+
+        if mode == 'rb+':
```

Comment:
Maybe not do this for now. If we do it, it should be `r+` or `rs+` in analogy with baseband.

Reply:
Removed